src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -478,61 +479,74 @@ def build_container_from_source_with_buildpack(self, image_name, source):  # pyl
         except Exception as ex:
             raise ValidationError(f"Unable to run 'pack config' command to set default builder: {ex}") from ex
 
-        # Run 'pack build' to produce a runnable application image for the Container App
-        command = [pack_exec_path, 'build', image_name, '--builder', builder_image_name, '--path', source]
-        buildpack_run_tag = get_latest_buildpack_run_tag("aspnet", "7.0")
-        if buildpack_run_tag is not None:
-            buildpack_run_image = f"mcr.microsoft.com/oryx/builder:{buildpack_run_tag}"
-            logger.debug(f"Determined the run image to use as {buildpack_run_image}.")
-            command.extend(['--run-image', buildpack_run_image])
-
         # If the user specifies a target port, pass it to the buildpack
         if self.target_port:
             command.extend(['--env', f"PORT={self.target_port}"])
 
-        logger.debug(f"Calling '{' '.join(command)}'")
-        try:
-            is_non_supported_platform = False
-            with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
+        could_build_image = False
+        for builder_image in builder_image_list:
+            # Run 'pack build' to produce a runnable application image for the Container App
+            # Specify the image as the 'build-cache' image to ensure that the local cache is used for build layers
+            command = [pack_exec_path, 'build', cache_image_name, '--builder', builder_image, '--path', source, '--tag', image_name]
 
-                # Stream output of 'pack build' to warning stream
-                while process.stdout.readable():
-                    line = process.stdout.readline()
-                    if not line:
-                        break
+            logger.debug(f"Calling '{' '.join(command)}'")
+            try:
+                is_non_supported_platform = False
+                is_non_supported_os = False
+                with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
 
-                    stdout_line = str(line.strip(), 'utf-8')
-                    logger.warning(stdout_line)
-                    if not is_non_supported_platform and "No buildpack groups passed detection" in stdout_line:
-                        is_non_supported_platform = True
+                    # Stream output of 'pack build' to warning stream
+                    while process.stdout.readable():
+                        line = process.stdout.readline()
+                        if not line:
+                            break
 
-                # Update the result of process.returncode
-                process.communicate()
-                if is_non_supported_platform:
-                    raise ValidationError("Current buildpacks do not support the platform targeted in the provided source code.")
+                        stdout_line = str(line.strip(), 'utf-8')
+                        logger.warning(stdout_line)
```

Comment:
I'm wondering how valuables are the logs before we've validated for sure that the current builder is the right one. I'm not feeling strongly, but would we lose much if we only started showing the logs once we know the builder will build?

Reply:
Discussed offline -- we will capture the stdout in a separate variable that we'll flush _only_ when the builder was successful in building the provided application source.